pynamodb/models.py

```diff
@@ -335,112 +328,34 @@ def __repr__(self):
-        :param value: The new value for the attribute.
-        :param action: The action to take if this item already exists.
-            See: http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateItem.html#DDB-UpdateItem-request-AttributeUpdate
-        """
-        warnings.warn("`Model.update_item` is deprecated in favour of `Model.update` now")
-
-        self._conditional_operator_check(conditional_operator)
-        args, save_kwargs = self._get_save_args(null_check=False)
-        attribute_cls = None
-        for attr_name, attr_cls in self.get_attributes().items():
-            if attr_name == attribute:
-                attribute_cls = attr_cls
-                break
-        if not attribute_cls:
-            raise ValueError("Attribute {0} specified does not exist".format(attr_name))
-        if save_kwargs.get(pythonic(RANGE_KEY)):
-            kwargs = {pythonic(RANGE_KEY): save_kwargs.get(pythonic(RANGE_KEY))}
-        else:
-            kwargs = {}
-        if len(expected_values):
-            kwargs.update(expected=self._build_expected_values(expected_values, UPDATE_FILTER_OPERATOR_MAP))
-        kwargs[pythonic(ATTR_UPDATES)] = {
-            attribute_cls.attr_name: {
-                ACTION: action.upper() if action else None,
-            }
-        }
-        if value is not None:
-            kwargs[pythonic(ATTR_UPDATES)][attribute_cls.attr_name][VALUE] = {
-                ATTR_TYPE_MAP[attribute_cls.attr_type]: attribute_cls.serialize(value)
-            }
-        kwargs[pythonic(RETURN_VALUES)] = ALL_NEW
-        kwargs.update(conditional_operator=conditional_operator)
         kwargs.update(condition=condition)
-        data = self._get_connection().update_item(
-            *args,
-            **kwargs
-        )
-
-        for name, value in data.get(ATTRIBUTES).items():
-            attr_name = self._dynamo_to_python_attr(name)
-            attr = self.get_attributes().get(attr_name)
-            if attr:
-                setattr(self, attr_name, attr.deserialize(attr.get_value(value)))
-        return data
+        return self._get_connection().delete_item(*args, **kwargs)
 
-    def update(self, attributes=None, actions=None, condition=None, conditional_operator=None, **expected_values):
+    def update(self, actions=None, condition=None):
         """
         Updates an item using the UpdateItem operation.
 
-        :param attributes: A dictionary of attributes to update in the following format
-                            {
-                                attr_name: {'value': 10, 'action': 'ADD'},
-                                next_attr: {'value': True, 'action': 'PUT'},
-                            }
+        :param actions:
         """
-        if attributes is not None and not isinstance(attributes, dict):
-            raise TypeError("the value of `attributes` is expected to be a dictionary")
         if actions is not None and not isinstance(actions, list):
```

Comment:
`actions` should not be None -- just check that it is a list and add a check that the list it isn't empty